Jenkinsfile

```diff
@@ -166,6 +166,12 @@ try {
             stage("Linting") {
               sh "scripts/lint.sh"
             }
+          }, nix: {
+            stage("nix-setup") {
+              sh "curl -L https://nixos.org/nix/install > nix-install.sh && chmod +x nix-install.sh && ./nix-install.sh --daemon --yes"
+              sh "/nix/var/nix/profiles/default/bin/nix-shell --pure --run \"mkdir nix-debug && cd nix-debug && cmake -GNinja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ .. && ninja && ./hyriseTest\""
+              sh "/nix/var/nix/profiles/default/bin/nix-shell --pure --run \"mkdir nix-release && cd nix-release && cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ .. && ninja && ./hyriseTest\""
```

Comment:
`-NOLTO=TRUE` should help here to limit compile time.

Reply:
And we should limit the cores ninja uses as it runs with other processes concurrently.

Reply:
I took over the `ninja all -j \$(( \$(nproc) / 5))` command from other stages, even if I have to admit that I do not understand it.

Reply:
Should I use `-NOLTO=TRUE` like that?

```
cmake -GNinja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -NOLTO=TRUE ..
```

Reply:
You might need to merge the master. `-DNOLTO=TRUE` is the full command (see here: 
https://github.com/hyrise/hyrise/blob/52985d5547be37eefacc2f922f27b5092cecb7be/Jenkinsfile#L117C23-L117C36).

`ninja all -j \$(( \$(nproc) / 5))` uses the `-j` argument to set the cores to a fifth of the cores available. Please set that to the number of parallel stages.

Reply:
But isn't that managed by the OS anyway? So if we run ninja with all cores, it will submit (#cores)*stages threads that run in parallel. If we submit exactly the number of threads that we also have cores for, we will not achieve every core working exclusively on a single thread (that is up to the OS). I fear there is optimization made that does increase quality of the build, just the technical depth.

I added it, but I am not sure if that is of benefit.

Reply:
There are multiple stages running in parallel. If you run several `ninja` which each run as many softwarethreads as hardware threads, you can be thrashing your system.

Reply:
I don't understand how this change is related to quality of a build or why it adds technical depth.

Reply:
But wouldn't that be balanced by the OS? We would crash the system, if there was more work incoming than can be processed, but that is not the case here.